### PR TITLE
chore(CPlusPlus): add reverse the string wordwise

### PR DESCRIPTION
**Pull request by RK-Shandilya**
**PR Checklist:**
- [x] My submission is formatted according to the guidelines in the [contributing guide](https://github.com/MakeContributions/DSA#contribution-guidelines)
- [ ] My addition is on refer on the language README.md file
- [x] My addition does not have a spelling problem
- [x] My submission has a proper and user-friendly description of the algorithm
- [x] My submission has the time complexity of the algorithm
- [x] My submission has sample input-output of the program (NOT FOR PYTHON)

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**

**Other information:**

### CODE REVIEW

**Comment by RK-Shandilya** [comment]
// Description :- Given a string, the task is to reverse the order of the words in the given string. 
// Example :-
// Input 1:
//    A = "the sky is blue"
// Input 2:
//    A = "this is ib"
// Output 1:
//    "blue is sky the"
// Output 2:
//    "ib is this"


// Time Complexity = O(N), Space Complexity = O(N)
string Solution::solve(string s) {
   vector<string>v;
        string str="";
        for(int i=0;i<s.length();i++){
            if(s[i]!=' '){
                str+=s[i];
            }
            else if(str!="" && s[i]==' '){
                v.push_back(str);
                str="";
            }
        }
        if(str!=""){
            v.push_back(str);
        }
        str="";
        for(int i=v.size()-1;i>0;i--){
                str+=v[i];
                str+=' ';
        }
        str+=v[0];
        return str;
}

**Comment by RK-Shandilya** [comment]
@Arsenic-ATG Could you please review and approve it?

**Comment by RK-Shandilya** [comment]
Can you give thorough review, I have updated my file please have a look and if everything is fine then please approve it.

File: algorithms/CPlusPlus/Strings/ReverseTheStringWordwise.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+//    A = "the sky is blue"
+// Input 2:
+//    A = "this is ib"
+// Output 1:
+//    "blue is sky the"
+// Output 2:
+//    "ib is this"
+
+
+// Time Complexity = O(N), Space Complexity = O(N)
```
Comment (Arsenic-ATG) [inline algorithms/CPlusPlus/Strings/ReverseTheStringWordwise.cpp:13]: include the necessary headers
```suggestion
// Time Complexity = O(N), Space Complexity = O(N)

#include <iostream>
#include <string>
#include <vector>
```

File: algorithms/CPlusPlus/Strings/ReverseTheStringWordwise.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+string Solution::solve(string s) {
```
Comment (Arsenic-ATG) [inline algorithms/CPlusPlus/Strings/ReverseTheStringWordwise.cpp:14]: the `solve` function seem to belong to `Solution` namespace which is not present in the code itself. 
either create a Solution namespace to fit this in or more preferably declare the function in global namespace.

```suggestion
string solve(string s) {
```

File: algorithms/CPlusPlus/Strings/ReverseTheStringWordwise.cpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+   vector<string>v;
```
Comment (Arsenic-ATG) [inline algorithms/CPlusPlus/Strings/ReverseTheStringWordwise.cpp:15]: the program extensively uses objects from standard namespace without explicitly specifying the system about the same. 

either specify that the object belong to standard namespace ( for example `std::vector<std::string> v:`, `std::cin` etc ) or simply tell the system that you are using standard namespace thought the program ( `using namespace std;`)

File: algorithms/CPlusPlus/Strings/ReverseTheStringWordwise.cpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+                str+=' ';
+        }
+        str+=v[0];
+        return str;
+}
+
+int main()
+{
+    string s;
+    cin>>s;
```
Comment (Arsenic-ATG) [inline algorithms/CPlusPlus/Strings/ReverseTheStringWordwise.cpp:41]: `std::cin` stops reading once a space is encountered, the program is meant to be taking words for proper demonstration/testing, consider using `std::getline ()` instead
```suggestion
    std::getline (std::cin, s);
```

**Comment by RK-Shandilya** [comment]
@Arsenic-ATG I have resolved all the issues mentioned. Please review it once again.

**Comment by Arsenic-ATG** [comment]
> @Arsenic-ATG I have resolved all the issues you mentioned. Please review it once again and it is running fine.

@RK-Shandilya nice, now all that is left is updating the corresponding README.md file here. 
If you face any issue or need help with something then feel free to ask here 🙂

**Comment by RK-Shandilya** [comment]
@Arsenic-ATG Sorry sir, but i don't know how to update readme.md file.

**Comment by Arsenic-ATG** [comment]
> @Arsenic-ATG Sorry sir, but i don't know how to update readme.md file.

not a problem, Here are the steps that I would follow in order to do that 
1. go to the base branch of the repository ( it's [RK-Shandilya:patch-2](https://github.com/RK-Shandilya/DSA/tree/patch-2) in your case )
Screenshot 2022-12-14 at 6 33 11 PM

2. navigate to [DSA/algorithm/CPlusPlus/README.md](https://github.com/RK-Shandilya/DSA/blob/patch-2/algorithms/CPlusPlus/README.md) and edit it ( clicking on the little pencil icon right there )
Screenshot 2022-12-14 at 6 36 33 PM

3. just add your program name and path ( as you might have already did in #1102 )
Screenshot 2022-12-14 at 6 39 12 PM

4. once the changes are done, write a commit message and commit the changes to your branch 
Screenshot 2022-12-14 at 6 41 24 PM

5. if done correctly, all the changes should automatically reflected back to this PR and you should be good to go 👍 

--- 

hope I was clear enough 
If you still have any doubt, please feel free to ask

**Comment by RK-Shandilya** [comment]
Done Sir and thankyou so much.

**Review by Arsenic-ATG — APPROVED** [review]

**Comment by RK-Shandilya** [comment]
@Arsenic-ATG Sir can you tell me what i have to do now??

**Comment by RK-Shandilya** [comment]
@ming-tsai please give write access to me so i can merge this pull request.

**Pull request merged**